cmd/flags/admin.go

```diff
@@ -70,13 +78,27 @@ func (s *AdminServer) setLogger(logger *zap.Logger) {
 // AddFlags registers CLI flags.
 func (s *AdminServer) AddFlags(flagSet *flag.FlagSet) {
 	flagSet.String(adminHTTPHostPort, s.adminHostPort, fmt.Sprintf("The host:port (e.g. 127.0.0.1%s or %s) for the admin server, including health check, /metrics, etc.", s.adminHostPort, s.adminHostPort))
+	tlsAdminHTTPFlagsConfig.AddFlags(flagSet)
 }
 
 // InitFromViper initializes the server with properties retrieved from Viper.
-func (s *AdminServer) initFromViper(v *viper.Viper, logger *zap.Logger) {
+func (s *AdminServer) initFromViper(v *viper.Viper, logger *zap.Logger) error {
 	s.setLogger(logger)
 
 	s.adminHostPort = v.GetString(adminHTTPHostPort)
+	if tlsAdminHTTP, err := tlsAdminHTTPFlagsConfig.InitFromViper(v); err == nil {
+		s.TLSConfig = tlsAdminHTTP
```

Comment:
nit: I don't think we need to keep the s.TLSConfig field. After it's parsed here, you can all Config(), and if no errors, then you get the `tlsCfg` pointer that can be checked for `nil` in Serve(), such that `TLSConfig` is never needed again. And I would assign `tlsCertWatcherCloser` here too, rather than later.

Reply:
Ok!